src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherNonresidentInformationValidation.java

```diff
@@ -326,33 +327,33 @@ public boolean validate(AttributedDocumentEvent event) {
             if (nonResidentAlienTax.getTaxSpecialW4Amount() != null) {
                 // Conditions to be met for "Special W-4 Amount" error to be generated
                 // income class code should be fellowship
-                if (!nonResidentAlienTax.getIncomeClassCode().equals(DisbursementVoucherConstants.NRA_TAX_INCOME_CLASS_FELLOWSHIP)) {
+                if (!nonResidentAlienTax.getIncomeClassCode().equals(DisbursementVoucherConstants.NONRESIDENT_TAX_INCOME_CLASS_FELLOWSHIP)) {
                     errors.putErrorWithoutFullErrorPath("DVNRATaxErrors",
-                            FPKeyConstants.ERROR_DV_NRA_TAX_WHEN_CHECKED_SHOULD_BE_SELECTED,
+                            FPKeyConstants.ERROR_DV_NONRESIDENT_TAX_WHEN_CHECKED_SHOULD_BE_SELECTED,
                             "Special W-4 Amount", "Income Class Code : Fellowship");
                     return false;
                 }
                 // Federal and State tax rate should be zero.
                 if (!(nonResidentAlienTax.getStateIncomeTaxPercent().equals(KualiDecimal.ZERO))
                         || !(nonResidentAlienTax.getFederalIncomeTaxPercent().equals(KualiDecimal.ZERO))) {
```

Comment:
Another spot that needs to use BigDecimal instead of KualiDecimal. This section also needs to be fixed in base code.

Reply:
updated